### ISSUE TITLE
[R][Docs] Add a non-technical introduction to the functioning of arrow

### ISSUE THREAD

**Issue by oliviermeslin** [apache/arrow#40547]
### Describe the enhancement requested

__TL;DR: I suggest to complement the `arrow` documentation with an introduction to the functioning of `arrow`, specifically designed for R users with limited computer science background. I wrote such an [introduction in French](https://www.book.utilitr.org/03_fiches_thematiques/fiche_arrow) ([link to automated English translation](https://www-book-utilitr-org.translate.goog/03_fiches_thematiques/fiche_arrow?_x_tr_sl=fr&_x_tr_tl=en&_x_tr_hl=fr&_x_tr_pto=wapp)); it's intentionally written in plain and sometimes imprecise language, avoiding most technical terms that usually puzzle newcomers. I'm ready to translate it to English, provided that the arrow team agrees with adding it to the documentation.__

## Context

My employer (a major European statistical organization) decided to move away from proprietary statistical softwares (mostly SAS) to embrace open source alternatives (mostly R and Python), and decided recently that data should be stored as Parquet files, making `arrow/dplyr` the standard approach to data processing when working with R.

I noticed repeatedly that `R` users who are new to `arrow` do not use `arrow` properly, because of an imperfect understanding of the way `arrow` works. For instance, they typically use `collect()` on large Parquet files (resulting in RAM saturation), because they do not understand the difference between `compute()` and `collect()`, or they write extremely long arrow/dplyr queries, resulting in session crashes.

Importantly, most of my colleagues have a strong background in statistics and data processing and an intermediate level in R, but a limited background in computer science.

## The arrow documentation

In my experience, this imperfect understanding comes from two causes:

- Some parts of the documentation are quite difficult to understand for newcomers because of unknown technical terms. For instance, my colleagues are often unable to understand the [following paragraph from the arrow documentation](https://arrow.apache.org/docs/r/) because they do not really know what the words interface, API or backend mean: "The arrow R package exposes an interface to the Arrow C++ library, enabling access to many of its features in R. It provides low-level access to the Arrow C++ library API and higher-level access through a [dplyr](https://dplyr.tidyverse.org/) backend and familiar R functions." As for myself, I struggled for months before understanding these notions.

- The [Apache Arrow R Cookbook](https://arrow.apache.org/cookbook/r/index.html) does not really give a general overview of the functioning of `arrow`. For instance, the differences between `compute()` and `collect()` and between the dplyr and acero execution engines, or the limitations of lazy evaluation are not really explained, although they are essential when working on large datasets.

Disclaimer: I want to stress that the point made here is not a criticism of the Arrow documentation; I just want to point that this documentation may not be well suited for newcomers with a limited background in computer science.

## My suggestion

I noticed that my colleagues were able to use `arrow` properly, once they were explained the functioning of `arrow` in plain, non-technical terms. That's why I wrote a [long and gentle introduction to `arrow` in French](https://www.book.utilitr.org/03_fiches_thematiques/fiche_arrow) ([link to automated English translation](https://www-book-utilitr-org.translate.goog/03_fiches_thematiques/fiche_arrow?_x_tr_sl=fr&_x_tr_tl=en&_x_tr_hl=fr&_x_tr_pto=wapp)), specifically designed for R users with limited computer science background. This introduction explains the functioning of `arrow` in plain language, avoiding most technical terms that usually puzzle newcomers. I explain important notions (for instance: lazy evaluation, execution engine) in intuitive and somewhat imprecise terms, so that newcomers get a first rough understanding of how things work with `arrow`, before switching to the current documentation.

I now think that this introduction could be a valuable contribution to the `arrow` documentation. I'm ready to translate/adapt it in English, for instance as a vignette for the `R` package, or as an overview chapter in the Apache Arrow R Cookbook. However, I'm very aware that this introduction could be considered as subpar, not precise enough or even misleading. That's why I would like to know if the arrow team is ready to consider this suggestion.

### Component(s)

Documentation, R

**Comment by amoeba** [comment]
Hi @oliviermeslin, thank you for filing this.

What you've learned from helping others learn how to be effective with the arrow R package is very welcome here. I think the structure you use in your guide is very useful and the content makes the most sense to me as a new vignette. There's [existing work](https://github.com/apache/arrow/issues/35875) an making the top-level readme more newcomer-friendly so maybe some of what you've already written could help with that work too.

I know @thisisnic and @stephhazlitt have spent a lot of time teaching arrow to others so their insight would be useful too.

**Comment by thisisnic** [comment]
I love this suggestion @oliviermeslin!  I think that you are much better placed than the experienced arrow devs (who are so exposed to this terminology that we no longer realise it's confusing to newcomers), and I'd be delighted to review any PRs you want to make to the arrow cookbook or vignettes.

I think the best way to proceed would be for you to go ahead with any pull requests you'd like to make, and we can address any concerns about precision about language in the discussion on the pull request (seeing as most PRs have a bit of back and forth before they're merged anyway!).

There is a PR open to update the README, so it might be better for this to have merged first so we can iterate on the improvements that have been made there so far: https://github.com/apache/arrow/pull/40148/

**Comment by oliviermeslin** [comment]
@thisisnic , @amoeba : thanks for your kind answers! I already started writing a vignette, I'll let you know when it's ready to be reviewed.

**Comment by oliviermeslin** [comment]
@amoeba, @thisisnic: I opened a PR.

**Comment by thisisnic** [comment]
I think this is a worthwhile task if anyone wants to pick this up, but for now I'm closing as nobody is looking at it.

**Comment by thisisnic** [comment]
This issue has been closed because there has been no activity for the 14 days since the stale warning. Feel free to reopen if this has been closed in error.
